### PR TITLE
Standarized the display of map properties

### PR DESCRIPTION
**Pull request by FuriousLlama**

### CODE REVIEW

File: frontend/src/features/properties/selector/utils.ts [hunk 1]
```diff
@@ -1,15 +1,56 @@
...
+  label: NameSourceType;
+  value: string;
+}
+
+export const getPropertyName = (property: IMapProperty): PropertyName => {
+  if (property.pid !== undefined && property.pid?.length > 0 && property.pid !== '0') {
+    return { label: NameSourceType.PID, value: pidFormatter(property.pid.toString()) };
+  } else if (property.pin !== undefined && property.pin?.length > 0 && property.pin !== '0') {
+    return { label: NameSourceType.PIN, value: property.pin.toString() };
+  } else if (property.planNumber !== undefined) {
```
Comment (devinleighsmith) [inline frontend/src/features/properties/selector/utils.ts:25]: assuming this is a string (I think it is), shouldn't this also check for non-zero length? also worth mentioning that !!property.pid checks for non-undefined, non-null, and also non-empty.

File: frontend/src/features/properties/selector/search/PropertySelectorSearchContainer.tsx [hunk 2]
```diff
@@ -189,7 +188,6 @@ export const featuresToIdentifiedMapProperty = (
           longitude: boundedCenter[0],
           legalDescription: feature?.properties?.LEGAL_DESCRIPTION,
         };
-        property.id = getPropertyIdentifier(property);
```
Comment (devinleighsmith) [inline frontend/src/features/properties/selector/search/PropertySelectorSearchContainer.tsx:192]: This seems like it could break something. Can you confirm that duplicate detection still works when attempting to add multiple of the same property?

File: frontend/src/features/properties/selector/search/PropertySearchSelectorFormView.test.tsx [hunk 3]
```diff
@@ -187,54 +189,35 @@ describe('PropertySearchSelectorFormView component', () => {
...
   describe('selecting results', () => {
     it('search results can be selected', async () => {
-      const { findByTestId } = setup({ searchResults: mockPropertyLayerSearchResponse.features });
-      const checkbox = await findByTestId(
-        'selectrow-WHSE_CADASTRE.PMBC_PARCEL_FABRIC_POLY_SVW.fid-674bf6f8_180d8c9b18e_7c12',
-      );
+      const { findByTestId, container } = setup({
+        searchResults: toMapProperty(mockPropertyLayerSearchResponse.features),
+      });
+      console.log(container.outerHTML);
```
Comment (devinleighsmith) [inline frontend/src/features/properties/selector/search/PropertySearchSelectorFormView.test.tsx:195]: can we remove this to avoid cluttering up the logs?

File: frontend/src/features/properties/selector/components/MapClickMonitor.tsx [hunk 4]
```diff
@@ -40,7 +40,7 @@ export const mapFeatureToProperty = (
     .getBounds()
     .getCenter();
   return {
-    pid: selectedFeature?.properties?.PID_NUMBER ?? undefined,
+    pid: selectedFeature?.properties?.PID ?? undefined,
```
Comment (devinleighsmith) [inline frontend/src/features/properties/selector/components/MapClickMonitor.tsx:43]: what is the rationale for this change?

File: frontend/src/features/properties/map/research/add/SelectedPropertyRow.tsx [hunk 5]
```diff
@@ -21,7 +21,28 @@ export const SelectedPropertyRow: React.FunctionComponent<ISelectedPropertyRowPr
   index,
   property,
 }) => {
-  const propertyIdentifier = getPropertyIdentifier(property);
+  const propertyName = getPropertyName(property);
+  let propertyIdentifier = '';
+  switch (propertyName.label) {
+    case NameSourceType.PID:
```
Comment (devinleighsmith) [inline frontend/src/features/properties/map/research/add/SelectedPropertyRow.tsx:27]: you can collapse all of the cases to reduce code duplication.

File: frontend/src/features/properties/map/research/add/SelectedPropertyRow.tsx [hunk 6]
```diff
@@ -21,7 +21,28 @@ export const SelectedPropertyRow: React.FunctionComponent<ISelectedPropertyRowPr
...
+      propertyIdentifier = `${propertyName.label}: ${propertyName.value}`;
+      break;
+    case NameSourceType.PIN:
+      propertyIdentifier = `${propertyName.label}: ${propertyName.value}`;
+      break;
+    case NameSourceType.PLAN:
+      propertyIdentifier = `${propertyName.label}: ${propertyName.value}`;
+      break;
+    case NameSourceType.LOCATION:
```
Comment (devinleighsmith) [inline frontend/src/features/properties/map/research/add/SelectedPropertyRow.tsx:36]: and then collapse these two.

**Review by devinleighsmith — APPROVED** [review]

File: frontend/src/features/properties/selector/search/PropertySelectorSearchContainer.tsx (hunk 2 above)
Comment (FuriousLlama) [reply frontend/src/features/properties/selector/search/PropertySelectorSearchContainer.tsx:192]: Yeah I checked that logic and updated the generation of id's in the table and in the selection as well as tests

File: frontend/src/features/properties/selector/search/PropertySearchSelectorFormView.test.tsx (hunk 3 above)
Comment (FuriousLlama) [reply frontend/src/features/properties/selector/search/PropertySearchSelectorFormView.test.tsx:195]: for sure!

File: frontend/src/features/properties/selector/components/MapClickMonitor.tsx (hunk 4 above)
Comment (FuriousLlama) [reply frontend/src/features/properties/selector/components/MapClickMonitor.tsx:43]: The definition of the IMapProperty defines the pid to be a string or undefined. Pid number was returning a number instead breaking the type promise for the interface and causing some issues downstream. i.e. checks for length of a string

Comment (FuriousLlama) [reply frontend/src/features/properties/selector/components/MapClickMonitor.tsx:43]: We should choose if we want to handle the pid as a string or a number in the frontend and abstract the conversions to the lowest level. That way we can have some guarantees when coding against that piece of info

File: frontend/src/features/properties/map/research/add/SelectedPropertyRow.tsx (hunk 5 above)
Comment (FuriousLlama) [reply frontend/src/features/properties/map/research/add/SelectedPropertyRow.tsx:27]: sure can do

File: frontend/src/features/properties/selector/utils.ts (hunk 1 above)
Comment (FuriousLlama) [reply frontend/src/features/properties/selector/utils.ts:25]: We could, but perhaps that is another conversation about the expectations for a plan number. Maybe empty string is valid? (very odd)
